Fix pruning in vendor script

dmcgowan opened the pull request:
Currently the vendor script prunes files which are currently checked in.

dmcgowan committed:
Fix pruning on tar split

Currently the vendor script prunes files which are currently checked in.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

dmcgowan commented:
ping @vbatts @icecrime

dmcgowan commented:
Oops noticed this picks up the LICENSE as well, does the vendor script need an update to not remove LICENSE files?

dmcgowan commented:
Added second commit which will keep the LICENSE files

ping @tianon for changes to vendor script

stevvooe commented:
LGTM

tiborvass commented:
LGTM

tianon commented on hack/.vendor-helpers.sh:
```diff
@@ -112,7 +112,7 @@ clean() {
 	unset IFS
 	for dir in "${prune[@]}"; do
-		find "$dir" -maxdepth 1 -not -type d -exec rm -v -f '{}' +
+		find "$dir" -maxdepth 1 -not -type d -not -name "LICENSE*" -exec rm -v -f '{}' +
```
Would you mind switching this to single quotes?  Also, do we want to include `COPYING*` as well, just in case? (not sure if any of our deps have one)

dmcgowan replied:
Yes I can switch to single, and sure we can add `COPYING*`, I am unaware of the significance of that file though

tianon replied:
Some projects use "COPYING" instead of "LICENSE", but I don't know that
it's very common in Go projects.

dmcgowan committed:
Keep license files

Currently the vendor script removes directories which do not have imported go packages, however this also ends up removing license files which may be other directories.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

dmcgowan commented:
Added `COPYING*` as well. It picked up a few files from toml which are not necessarily relevant since they are in subdirectories without vendored code. I would say it is better to included licenses for pruned code than omit licenses for used code. A smarter implementation would probably only include files which have files remaining in the current or sub directories.

tianon commented:
Nice, thanks (I agree, better to have some false-positives in this case)!  LGTM

jfrazelle commented:
LGTM

vbatts commented:
<3
